packages/gestalt/src/Badge.tsx

```diff
@@ -71,52 +82,123 @@ export default function Badge({ position = 'middle', text, type = 'info', toolti
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
+  const COLOR_TEXT_MAP = Object.freeze({
+    'info': 'shopping',
+    'error': 'error',
+    'warning': 'warning',
+    'success': 'success',
+    'neutral': isInVRExperiment ? 'default' : 'inverse',
+    'recommendation': 'recommendation',
+    'darkWash': 'light',
+    'lightWash': 'dark',
+  });
+
   let styleType: TypeOptions | InteractiveTypeOptions = type;
 
   if (shouldUseTooltip) {
     styleType = `interactive-${type}`;
   }
 
-  const csBadge = cx(styles.Badge, styles[position], styles[styleType]);
+  const { handleOnBlur, handleOnFocus, handleOnMouseEnter, handleOnMouseLeave, isFocused } =
+    useInteractiveStates();
+
+  const cxStyles = useMemo(
+    () =>
+      cx(styles.badge, styles[styleType], {
+        [styles.middle]: !shouldUseTooltip && position === 'middle',
+        [styles.top]: !shouldUseTooltip && position === 'top',
+        [styles.focusInnerBorder]:
+          isInVRExperiment &&
+          isFocused &&
+          isFocusVisible &&
+          !['darkWash', 'lightWash'].some((color) => color === type),
+        [styles.focusInnerBorderLight]:
+          isInVRExperiment && isFocused && isFocusVisible && type === 'darkWash',
+        [styles.focusInnerBorderDark]:
+          isInVRExperiment && isFocused && isFocusVisible && type === 'lightWash',
+      }),
+    [isFocused, isInVRExperiment, styleType, position, shouldUseTooltip, isFocusVisible, type],
+  );
+
+  const cxPositionStyles = useMemo(
+    () =>
+      cx({
+        [styles.middle]: position === 'middle',
+        [styles.top]: position === 'top',
+      }),
+    [position],
+  );
 
   const badgeComponent = (
-    <div className={csBadge}>
-      <Flex alignItems="center" gap={1}>
-        {shouldUseTooltip ? (
-          <Box aria-hidden>
-            <Icon
-              accessibilityLabel=""
-              color="inverse"
-              icon={ICON_MAP[type] as ComponentProps<typeof Icon>['icon']}
-              inline
-              size="14"
-            />
-          </Box>
-        ) : null}
-        <Box dangerouslySetInlineStyle={{ __style: { marginTop: '2px' } }} display="inlineBlock">
-          {text}
+    <Flex alignItems="center" gap={{ row: 1, column: 0 }}>
+      {shouldUseTooltip ? (
+        <Box alignContent="center" display="flex">
+          <Icon
+            accessibilityLabel=""
+            color={isInVRExperiment || type.endsWith('Wash') ? COLOR_ICON_MAP[type] : 'inverse'}
+            icon={ICON_MAP[type] as ComponentProps<typeof Icon>['icon']}
+            inline
+            size={isInVRExperiment ? '12' : '14'}
+          />
         </Box>
-      </Flex>
-    </div>
+      ) : null}
+      <Box alignContent="center" display="flex">
+        <Text
+          color={isInVRExperiment || type.endsWith('Wash') ? COLOR_TEXT_MAP[type] : 'inverse'}
```

Comment:
this conditional feels odd, can `Color_Text_Map` cover all the cases? 

it seems like `type` can never be undefined right - since it's always assigned a default value?

Reply:
this will eventuallybe cleaned up when we merge. type.endsWith('Wash') when outside of VR cant be inversed.